vllm/compilation/wrapper.py

```diff
@@ -82,6 +83,13 @@ def bytecode_hook(self, old_code: CodeType, new_code: CodeType):
 
         self.compiled_codes.append(new_code)
 
+        if self.vllm_config.compilation_config.use_cudagraph and \
+            "update" in new_code.co_names:
+            import depyf
+            src = depyf.decompile(new_code)
+            msg = "Assigning / modifying buffers of nn.Module during forward pass is not allowed when using cudagraph inside the compiler because it will cause silent errors. Please use eager mode or fix the code. The following code contains clues about which buffer is being modified (please search for the usage of the function `update`):\n" + src  # noqa
+            raise RuntimeError(msg)
```

Comment:
Thanks for adding this preventive check!

For reference, I ran this with the old phi3 rotary embedding and the error message is

```
ERROR 12-29 07:19:20 core.py:200] torch._dynamo.exc.InternalTorchDynamoError: RuntimeError: Assigning / modifying buffers of nn.Module during forward pass is not allowed when using cudagraph inside the compiler because it will cause silent errors. Please use eager mode or fix the code. The following code contains clues about which buffer is being modified (please search for the usage of the function `update`):
ERROR 12-29 07:19:20 core.py:200] def forward(self, input_ids, positions, kv_caches, attn_metadata, intermediate_tensors, inputs_embeds):
ERROR 12-29 07:19:20 core.py:200]     graph_out_0 = __compiled_fn_1(inputs_embeds.size(0), inputs_embeds, self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['0']._modules['input_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['0']._modules[
ERROR 12-29 07:19:20 core.py:200]         'self_attn']._modules['qkv_proj']._parameters['weight'], positions.size
ERROR 12-29 07:19:20 core.py:200]         (0), positions, self._modules['layers']._modules['0']._modules[
ERROR 12-29 07:19:20 core.py:200]         'self_attn']._modules['rotary_emb']._buffers['long_short_cos_sin_cache'
ERROR 12-29 07:19:20 core.py:200]         ], kv_caches[0], self._modules['layers']._modules['0']._modules[
ERROR 12-29 07:19:20 core.py:200]         'self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['0']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['0']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['0']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['1']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['1']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[1], self._modules['layers']._modules['1']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['1']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['1']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['1']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['2']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['2']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[2], self._modules['layers']._modules['2']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['2']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['2']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['2']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['3']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['3']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[3], self._modules['layers']._modules['3']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['3']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['3']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['3']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['4']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['4']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[4], self._modules['layers']._modules['4']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['4']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['4']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['4']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['5']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['5']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[5], self._modules['layers']._modules['5']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['5']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['5']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['5']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['6']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['6']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[6], self._modules['layers']._modules['6']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['6']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['6']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['6']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['7']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['7']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[7], self._modules['layers']._modules['7']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['7']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['7']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['7']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['8']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['8']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[8], self._modules['layers']._modules['8']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['8']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['8']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['8']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['9']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['9']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[9], self._modules['layers']._modules['9']._modules
ERROR 12-29 07:19:20 core.py:200]         ['self_attn']._modules['o_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['9']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['9']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['9']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['10']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['10']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[10], self._modules['layers']._modules['10'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['10']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['10']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['10']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['11']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['11']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[11], self._modules['layers']._modules['11'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['11']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['11']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['11']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['12']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['12']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[12], self._modules['layers']._modules['12'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['12']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['12']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['12']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['13']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['13']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[13], self._modules['layers']._modules['13'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['13']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['13']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['13']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['14']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['14']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[14], self._modules['layers']._modules['14'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['14']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['14']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['14']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['15']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['15']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[15], self._modules['layers']._modules['15'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['15']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['15']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['15']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['16']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['16']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[16], self._modules['layers']._modules['16'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['16']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['16']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['16']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['17']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['17']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[17], self._modules['layers']._modules['17'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['17']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['17']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['17']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['18']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['18']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[18], self._modules['layers']._modules['18'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['18']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['18']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['18']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['19']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['19']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[19], self._modules['layers']._modules['19'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['19']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['19']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['19']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['20']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['20']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[20], self._modules['layers']._modules['20'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['20']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['20']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['20']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['21']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['21']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[21], self._modules['layers']._modules['21'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['21']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['21']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['21']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['22']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['22']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[22], self._modules['layers']._modules['22'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['22']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['22']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['22']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['23']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['23']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[23], self._modules['layers']._modules['23'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['23']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['23']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['23']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['24']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['24']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[24], self._modules['layers']._modules['24'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['24']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['24']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['24']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['25']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['25']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[25], self._modules['layers']._modules['25'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['25']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['25']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['25']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['26']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['26']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[26], self._modules['layers']._modules['26'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['26']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['26']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['26']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['27']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['27']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[27], self._modules['layers']._modules['27'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['27']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['27']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['27']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['28']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['28']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[28], self._modules['layers']._modules['28'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['28']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['28']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['28']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['29']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['29']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[29], self._modules['layers']._modules['29'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['29']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['29']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['29']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['30']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['30']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[30], self._modules['layers']._modules['30'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['30']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['30']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['30']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['31']._modules[
ERROR 12-29 07:19:20 core.py:200]         'input_layernorm']._parameters['weight'], self._modules['layers'].
ERROR 12-29 07:19:20 core.py:200]         _modules['31']._modules['self_attn']._modules['qkv_proj']._parameters[
ERROR 12-29 07:19:20 core.py:200]         'weight'], kv_caches[31], self._modules['layers']._modules['31'].
ERROR 12-29 07:19:20 core.py:200]         _modules['self_attn']._modules['o_proj']._parameters['weight'], self.
ERROR 12-29 07:19:20 core.py:200]         _modules['layers']._modules['31']._modules['post_attention_layernorm'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['layers']._modules['31']._modules[
ERROR 12-29 07:19:20 core.py:200]         'mlp']._modules['gate_up_proj']._parameters['weight'], self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['31']._modules['mlp']._modules['down_proj'].
ERROR 12-29 07:19:20 core.py:200]         _parameters['weight'], self._modules['norm']._parameters['weight'])
ERROR 12-29 07:19:20 core.py:200]     self._modules['layers']._modules['0']._modules['self_attn']._modules[
ERROR 12-29 07:19:20 core.py:200]         'rotary_emb']._non_persistent_buffers_set.clear()
ERROR 12-29 07:19:20 core.py:200]     self._modules['layers']._modules['0']._modules['self_attn']._modules[
ERROR 12-29 07:19:20 core.py:200]         'rotary_emb']._non_persistent_buffers_set.update({'long_cos_sin_cache',
ERROR 12-29 07:19:20 core.py:200]         'short_cos_sin_cache', 'long_short_cos_sin_cache'})
ERROR 12-29 07:19:20 core.py:200]     self._modules['layers']._modules['0']._modules['self_attn']._modules[
ERROR 12-29 07:19:20 core.py:200]         'rotary_emb']._buffers.clear()
ERROR 12-29 07:19:20 core.py:200]     self._modules['layers']._modules['0']._modules['self_attn']._modules[
ERROR 12-29 07:19:20 core.py:200]         'rotary_emb']._buffers.update({'short_cos_sin_cache': self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['0']._modules['self_attn']._modules['rotary_emb'].
ERROR 12-29 07:19:20 core.py:200]         _buffers['short_cos_sin_cache'], 'long_cos_sin_cache': self._modules[
ERROR 12-29 07:19:20 core.py:200]         'layers']._modules['0']._modules['self_attn']._modules['rotary_emb'].
ERROR 12-29 07:19:20 core.py:200]         _buffers['long_cos_sin_cache'], 'long_short_cos_sin_cache': graph_out_0[1]}
ERROR 12-29 07:19:20 core.py:200]         )
ERROR 12-29 07:19:20 core.py:200]     return graph_out_0[0]
ERROR 12-29 07:19:20 core.py:200] 
ERROR 12-29 07:19:20 core.py:200] 
ERROR 12-29 07:19:20 core.py:200] 
ERROR 12-29 07:19:20 core.py:200] You can suppress this exception and fall back to eager by setting:
ERROR 12-29 07:19:20 core.py:200]     import torch._dynamo
ERROR 12-29 07:19:20 core.py:200]     torch._dynamo.config.suppress_errors = True
ERROR 12-29 07:19:20 core.py:200] 
ERROR 12-29 07:19:20 core.py:200] 
```
I think the instruction is reasonably clear!